scanner/src/test/kotlin/scanners/fossid/FossIdTest.kt

```diff
@@ -643,7 +643,7 @@ class FossIdTest : WordSpec({
                 service.checkDownloadStatus(USER, API_KEY, scanCode)
             }
 
-            coVerify(exactly = 0) {
+            coVerify(exactly = 1) {
```

Comment:
Doesn't this now check exactly the same function signature as the check you added in the previous commit? And if so, can't we just drop this check completely?

Reply:
This now checks that there is one invocation in total. It matches the parameters exactly, but matches arbitrary values for the options. So, to put the assertion in words: "Assert that there is no invocation besides the one asserted exactly below." E.g. the one with:

```
 "auto_identification_detect_declaration" to "0",
 "auto_identification_detect_copyright" to "0"`
```

The reason why I added the assertion was that I also had a hard time figuring out why the test started failing.
And I thought the added assertion brings at least a bit of clarity / understanding.

Reply:
I read up a bit on this topic and it seems that what I've been doing with two assertions should be done with just one, using `verifyAll`, see https://github.com/mockk/mockk/issues/74.

Reply:
Hm, `verifyAll` also is not really what fits as that would verify all interactions with the mock, while my assertion is about all calls to the `runScan` function of the mock, but doesn't care about all the other function invocations.

Reply:
Any preference what to do @sschuberth ? I tend to change amend my commit which introduces the new assertion to remove the existing one above. (Rationale: we verify that the created scan is run, but don't check that it is not run twice. This seems reasonable as, like the test name suggests, the test is about creation of the scan - not about executing (running). Makes sense to you as well?

Reply:
I've updated the commits according to my previous comment.

Reply:
> Makes sense to you as well?

Yes. However, in the commit message shouldn't "without empty `options`" say "with empty `options`"?

Reply:
...right!